commons/src/main/java/org/infinispan/commons/marshall/jboss/ExtendedRiverUnmarshaller.java

```diff
@@ -46,4 +48,11 @@ public void finish() throws IOException {
       }
    }
 
+   int instanceCacheResizeCount() {
+      List instanceCache = (List) ReflectionUtil
+         .getValue(this, "instanceCache");
+
+      return instanceCache.size();
```

Comment:
I think `size()` will always return 0 at this point, because the instance cache is cleared between invocations. OTOH `ArrayList` has a `trimToSize()` method, maybe we could use that and keep using the same `RiverUnmarshaller` instance.